plugins/out_s3/s3.c

```diff
@@ -1832,11 +1908,13 @@ static void cb_s3_upload(struct flb_config *config, void *data)
         m_upload = mk_list_entry(head, struct multipart_upload, _head);
         complete = FLB_FALSE;
 
-        if (m_upload->complete_errors >= MAX_UPLOAD_ERRORS) {
+        if (m_upload->complete_errors > ctx->ins->retry_limit) {
             flb_plg_error(ctx->ins,
-                          "Upload for %s has reached max completion errors, "
+                          "Multipart Upload for %s has reached max "
+                          "s3:CompleteMultipartUpload errors, "
```

Comment:
nit: since we are checking this against user configured retry_limit may be we should make the message clear about that. 

like:

```
Multipart Upload for %s has failed s3:CompleteMultipartUpload more than configured retry_limit, output will give up
```

Reply:
updated